### ISSUE TITLE
[omnibus] geos package build error on RHEL-6

### ISSUE THREAD

**Issue by santisaez** [CartoDB/omnibus-cartodb#13]
I'm getting the following error when building `geos` package on a RHEL-6 system:

``` bash
lt-XMLTester: GeometryComponentFilter.cpp:34: virtual void geos::geom::GeometryComponentFilter::filter_ro(const geos::geom::Geometry*): Assertion `0' failed.
```

`make check` is step that is generating the error above:

https://github.com/CartoDB/omnibus-cartodb/blob/master/config/software/geos.rb#L16

Quoting from https://trac.osgeo.org/geos/ticket/535:

> `gcc` (at least some versions, I haven't checked new versions) will not create a virtual table for a class which has overridden virtual functions unless there is at least one non-inline function. The class `LinearComponentExtractor?` is defined completely in a header file with all implicit inline functions, thus a virtual table is not generated and the base class functions are called. The base class functions do nothing except assert.
> 
> To fix this at least one of the functions should be moved into a cpp file. It may work if you just define the function later in the header after the class declaration but I'm not sure. I also think there may be other classes like this in other spots in geos.

As shown in the [GEOS ticket](https://trac.osgeo.org/geos/ticket/535), another solution is to explicitly use `CFLAGS="-O1" CXXFLAGS="-O1"` options when calling `./configure` build script, this will fix the issue on RHEL-6 based systems.

**Comment by hsato42** [comment]
Do you have a PR I could merge?

**Comment by santisaez** [comment]
@hsato42 yes, this is the PR https://github.com/CartoDB/omnibus-cartodb/pull/14 that fixes the GEO build issue on RHEL-6.5, thanks!

**Comment by hsato42** [comment]
^merged
